### DISCUSSION TITLE
PID input settings.

### DISCUSSION THREAD

**Discussion by Tapap84** [Jyers/Marlin#444] [Q&A] [answered]
The new built in PID is an amazing feature. Is there a way to input old saved values or do you have to run a new PID every time you update? I wrote down my values.

**Comment by Jyers** [comment]
Currently the best way would be to do it through a console, but a menu for that might be a good idea. I'll look into adding one

**Reply by aryajuanda** [reply]
Wow nice one... why dont you create a pull request?

**Reply by Jyers** [reply]
@tititopher68 Sorry, I thought I had responded already. I've got a few things above it on my list, but I'm going to add some of your features soon.

**Comment by melvinmicallef** [comment]
I figured that putting them in the configuration files is easier if you know how to compile yourself :)

**Reply by G4GNX** [reply]
I do that too. I also have all of my other data in the config. Saves me having to look them up. :-)

**Comment by Timbo200** [comment]
If you use Octoprint I've found the EEPROM editor to be really useful.

https://plugins.octoprint.org/plugins/eeprom_marlin/

It basically takes the output from M503 and stores the values it retrieves.   After you flash the new firmware you can upload those same values back.

**Reply by Jyers** [reply]
Wait this is amazing! I might have to add this to the octoprint page on the wiki. 😄

**Comment by Tapap84** [comment]
just got back to a pc. replying on mobile doesn't work for me. man you guys do awesome work. got me so hooked i check the release section everyday for updates even though i cant print right now because of house repairs

**Comment by Jyers** [comment, marked as answer]
Just implemented this today. If you compile your own version you can get it now, otherwise it will be in the next release.

**Reply by Jyers** [reply]
@tititopher68 I don't know what you want me to say here. The only time someone get's co-authored on a commit is when I directly merge their pull request. Unfortunately I could not do that with yours because it has many changes I did not want to implement all at once. I also did not use any of your code when implementing the menu as I had a slightly different approach to the value handling. I honestly didn't even consider that you would want to be directly quoted in the commit. I really appreciate all the work you've done both in the community and with your additions. If it's that important to you to be directly mentioned in any commits derived from your code/ideas I will try my best to do that in the future. No need to be passive aggressive, I'm doing my best here.

**Reply by Jyers** [reply]
@tititopher68 I was just noting on how GitHub handles co-authoring. I know you're not super familiar with GitHub. GitHub automatically sets up co-authorship between me and the PR creator when I merge it. I don't actively mention users, its just the way GitHub manages it, hence why you were not mentioned since I wasn't merging a PR. Also glad to hear your working on a new project. I would recommend for future PRs to avoid starting with your heavily modified version and to add new settings to the current source. That way I can merge them as PRs rather than having to implement them myself. Thanks again for all your work.
